lib/chef/config.rb

```diff
@@ -49,5 +61,24 @@ class Config
       evt_loggers
     end
 
+    # Override the default values that were set by Ohai.
+    #
+    # REMOVEME once these configurables are removed from the top level of Ohai.
+    default :log_level, LOG_LEVEL
+    default :log_location, LOG_LOCATION
+
+    # Ohai::Config[:log_level] is deprecated and warns when set. Unfortunately,
+    # there is no way to distinguish between setting log_level and setting
+    # Ohai::Config[:log_level]. Since log_level and log_location are used by
+    # chef-client and other tools (e.g., knife), we will mute the warnings here
+    # by redefining the config_attr_writer to not warn for these options.
+    #
+    # REMOVEME once the warnings for these configurables are removed from Ohai.
```

Comment:
In the other case, it was that there is no way to distinguish between `--secret SECRET` on the CLI and `knife[:secret] = SECRET` in knife.rb after configs have been merged. The CLI secret option was redefined to write to a different key so `Chef::Config.knife[:secret]` could map to `knife[:secret]` and `Chef::Config.knife[:cl_secret]` could map to `--secret`.

In this case, the trouble is with multiple definitions of `default :log_level, LEVEL`. The ohai config file is the last writer of this configuration option, and therefore the winner. I thought about naming the configurable `:ohai_log_level` in `Ohai::Config` and redefining the `get` method for `:log_level` so that it will return `:ohai_log_level` if `log_level` isn't set in the config.rb. However, this method will get applied to `Chef::Config` too, since the `::Config` part of `Ohai::Config` and `Chef::Config` is really just `ChefConfig::Config`. Then we're back in essentially the same boat.